odl/discr/discretization.py

```diff
@@ -316,6 +316,26 @@ def __setitem__(self, indices, values):
         else:
             self.ntuple.__setitem__(indices, values)
 
+    @property
+    def restriction(self):
+        """The restriction operator associated with this vector.
+
+        See Also
+        --------
+        RawDiscretization.restriction : For full description
+        """
+        return self.space.restriction(self.ntuple)
+
+    @property
+    def extension(self):
+        """The extension operator associated with this vector.
+
+        See Also
+        --------
+        RawDiscretization.extension : For full description
+        """
+        return self.space.extension(self.ntuple)
```

Comment:
I personally find this quite similar to our `adjoint` syntax. Most guys are quick, but some could end up slow. The gains of the syntax is kindoff worth it anyway.